### DISCUSSION TITLE
Helmfile rendering secrets stored in Kubernetes secrets

### DISCUSSION THREAD

**Discussion by bersalazar** [helmfile/helmfile#1178] [Questions] [answered]
From helmfile, I'd like to do a lookup of a secret stored in the target Kubernetes cluster (as a Kubernetes secret) and into the helmfile template.

I'm aware of Helm's lookup function https://helm.sh/docs/chart_template_guide/functions_and_pipelines/#using-the-lookup-function which has this functionality, but I'd like to do the same when a helmfile template is rendered.

I'm also aware of https://github.com/helmfile/vals for doing remote secret store lookups, but this isn't an option. The secrets are stored in as a Kubernetes secret.
 
A possible solution is using helmfile's `exec` function with a kubectl command, but ideally I don't use this, as it complicates the template with the inline kubectl command.

Is there any alternative to doing this?

**Comment by yxxhero** [comment]
@bersalazar PR is welcome.

**Reply by bersalazar** [reply, marked as answer]
PR currently in review: https://github.com/helmfile/vals/pull/186
